shared/ghutil/client.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// client.go defines clients, and provide helpers overcome Github rate limit errors
+
+package ghutil
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+const (
+	maxRetryCount = 5
+	tokenReserve  = 50
+)
+
+var (
+	ctx = context.Background()
+)
+
+// GithubClientInterface contains a set of functions for Github operations
+type GithubClientInterface interface {
```

Comment:
rename this to GithubOperations? I dont think we need to add interface siffux

Reply:
done